src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/FluxCommunication.hpp

```diff
@@ -144,12 +149,16 @@ struct ReceiveDataForFluxes {
 
     const auto& inbox =
         tuples::get<typename flux_comm_types::FluxesTag>(inboxes);
-
+    const auto& self_id = db::get<Tags::Element<volume_dim>>(box).id();
     const auto& local_next_temporal_id = db::get<Tags::Next<temporal_id>>(box);
     const auto& mortars_next_temporal_id =
         db::get<Tags::Mortars<Tags::Next<temporal_id>, volume_dim>>(box);
     for (const auto& mortar_id_next_temporal_id : mortars_next_temporal_id) {
       const auto& mortar_id = mortar_id_next_temporal_id.first;
+      // If on an external boundary
+      if (mortar_id.second == self_id) {
```

Comment:
You can't use `self_id` to mark external boundaries.  An element can legitimately be its own neighbor.

Reply:
Yay for periodic single elements... The edge case that has always been a pain... If I recall correctly the suggestion was made to use something like half of max `size_t`.